docs/models/blockstates/introduction.md

```diff
@@ -1,9 +1,7 @@
 Introduction to Blockstate JSONs
 ================================
 
-Blockstate JSONs tell the game which model it should use depending on the values of the block's [blockstate properties][blockstate].
-A simple block with no properties only has a "*normal*" blockstate which is its default.
-A more complex block that can be displayed in different ways has so called *variants*.
+Blockstate JSONs are Minecraft's way to map "variant strings" in models. A variant string can be absolutely anything, from "inventory" to "power=5" to "I am your father." When the game searches for a model corresponding to a block in the world, it takes the [blockstate] for that position, and then it uses that to find the corresponding blockstate JSON and the variant string within it. The default is to use the block's registry name as the location of the blockstate JSON (block `examplemod:testblock` goes to JSON `examplemod:testblock` goes to filepath `assets/examplemod/blockstates/testblock.json`), and the variant string is pieced together from the blockstate's properties. More information can be found [here][statemapper]. The combination of a resource location (like the path to a blockstate JSON, e.g. `examplemod:blockstate`) and a variant string is represented with a `ModelResourceLocation`.
```

Comment:
I just tested setting the variant string to `Luke, I AM your father!`, and `"/;:',[]{}()*&^%$#@!<>?|\..`, and `�TBg���p�`. I pulled the last one straight out of `/dev/urandom`. All 3 worked just fine on items, so if there is a limit to that side, I'm not hitting it. On the other hand, whatever is registering block models lowercases the string first, so there's that.

I set the item to a variant in a blockstate with setCustomMRL, and I set the block's with a setCustomStateMapper.